### PR TITLE
feat: ripped out rollbar library, replaced with api invocation

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: src/logic/process-logs.js [hunk 1]
```diff
@@ -69,23 +70,29 @@ const getToLog = async (resultParsed, rb) => {
...
+      await Promise.all([
+        s3.putGzipObject(
+          process.env.LOG_STREAM_BUCKET_NAME,
+          `${resultParsed.logGroup.slice(1)}/${year}/${month}/${day}/${logLevel}-${logEvent.id}.json.gz`,
+          JSON.stringify(processedLogEvent)
+        ),
+        rb({
+          level: logLevel,
+          message: processedLogEvent.message,
+          timestamp: Math.floor(processedLogEvent.timestamp / 1000)
```
Comment (MacMcIrish) [inline src/logic/process-logs.js]: We can provide additional context here too, such as the log group / log stream

File: src/logic/util/rollbar.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const request = require('request-promise');
+
+module.exports = ({
+  level,
+  message,
+  timestamp = Math.floor(new Date() / 1000)
+}) => {
+  if (process.env.VERBOSE === '1') {
+    // eslint-disable-next-line no-console
+    console.log(message);
```
Comment (MacMcIrish) [inline src/logic/util/rollbar.js]: To support old usage, I'm good to remove if you are @simlu

File: src/logic/util/s3.js [hunk 3]
```diff
@@ -14,28 +15,35 @@ module.exports.putGzipObject = (bucket, key, data) => aws.call('s3:putObject', {
...
-module.exports.emptyBucket = (objParams, logger) => {
-  logger.info(`emptyBucket(): ${JSON.stringify(objParams)}`);
-  return listObjectsV2(objParams).then((result) => {
-    if (result.Contents.length === 0) {
-      return Promise.resolve();
+module.exports.emptyBucket = async (objParams,) => {
+  await rb({
+    level: 'info',
+    message: `emptyBucket(): ${JSON.stringify(objParams)}`
+  });
```
Comment (MacMcIrish) [inline src/logic/util/s3.js]: Changed from promise chaining to async await.
Only differences are the `await rb` lines

File: test/lambda/tests/processLogs/rollbarWithXray.spec.json [hunk 4]
```diff
@@ -1,7 +1,9 @@
 {
   "handler": "processLogs",
+  "timestamp": 1559595312,
   "env": {
-    "ROLLBAR_ACCESS_TOKEN": "rollbar_dummy_token"
+    "ROLLBAR_ACCESS_TOKEN": "rollbar_dummy_token",
+    "VERBOSE": "0"
```
Comment (MacMcIrish) [inline test/lambda/tests/processLogs/rollbarWithXray.spec.json]: For branch coverage

File: src/logic/empty-bucket.js [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
     if (Bucket === undefined) {
       return reject(new Error('No Bucket Provided.'));
     }
-    return s3.emptyBucket({ Bucket }, rb)
-      .then(() => {
-        rb.info(`${Bucket} emptied!`);
-        resolve();
-      }).catch(reject);
+    return s3.emptyBucket({ Bucket })
+      .then(() => rb({ level: 'info', message: `${Bucket} emptied!` }))
```
Comment (simlu) [inline src/logic/empty-bucket.js]: lambda-monitor-logger?

File: src/logic/util/rollbar.js (hunk 2 above)
Comment (simlu) [reply src/logic/util/rollbar.js]: yep rm

File: src/logic/util/rollbar.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        data: {
+          level,
+          environment: process.env.ENVIRONMENT,
+          body: {
+            message: {
+              body: message
+            }
+          },
+          timestamp,
+          fingerprint: message.split('\n')[0]
```
Comment (simlu) [inline src/logic/util/rollbar.js]:
```suggestion
          fingerprint: message.split('\n')[0]
```

crypto md5 hash

File: test/logic/process-logs.spec.js [hunk 7]
```diff
@@ -1,11 +1,28 @@
...
+      logs.push(message);
+    };
+    assert(process.env.VERBOSE === undefined);
+    process.env.VERBOSE = '1';
+  });
+
+  afterEach(() => {
+    console.log = consoleLogOriginal;
+    delete process.env.VERBOSE;
+  });
```
Comment (simlu) [inline test/logic/process-logs.spec.js]: use node-tdd recorder

File: src/handler.js [hunk 8]
```diff
@@ -1,27 +1,16 @@
...
 const emptyBucket = require('./logic/empty-bucket');
 
-const callbackify = (fn) => (event, context, rb) => new Promise((resolve, reject) => fn(
+const callbackify = (fn) => (event, context) => new Promise((resolve, reject) => fn(
   event,
   context,
-  (err, resp) => (err ? reject(err) : resolve(resp)),
-  rb
+  (err, resp) => (err ? reject(err) : resolve(resp))
 ));
```
Comment (simlu) [inline src/handler.js]: rm?

File: src/logic/process-logs.js [hunk 9]
```diff
@@ -1,13 +1,43 @@
...
 const promiseComplete = require('./util/promise-complete');
 const logz = require('./services/logz');
 const loggly = require('./services/loggly');
 const datadog = require('./services/datadog');
 
+const postToRollbar = ({
+  level,
+  message,
+  timestamp
+}) => (get(process, 'env.ROLLBAR_ACCESS_TOKEN', '') === ''
```
Comment (simlu) [inline src/logic/process-logs.js]: simplify

File: src/logic/empty-bucket.js [hunk 10]
```diff
@@ -1,19 +1,21 @@
 const get = require('lodash.get');
+const { logger } = require('lambda-monitor-logger');
 const cfnResponse = require('./util/cfn-response-wrapper');
 const s3 = require('./util/s3');
 
-module.exports = cfnResponse.wrap((event, context, callback, rb) => new Promise((resolve, reject) => {
+module.exports = cfnResponse.wrap((event, context, callback) => new Promise((resolve, reject) => {
```
Comment (simlu) [inline src/logic/empty-bucket.js]: no need to use new Promise here I guess

File: src/logic/process-logs.js (hunk 1 above)
Comment (simlu) [reply src/logic/process-logs.js]: Yeah, that would be great. Ideally we do it the same way as the old system where you could click the link to get to the correct log group

File: src/logic/util/s3.js [hunk 11]
```diff
@@ -14,28 +15,26 @@ module.exports.putGzipObject = (bucket, key, data) => aws.call('s3:putObject', {
   Body: zlib.gzipSync(data, { level: 9 })
 });
 
-module.exports.emptyBucket = (objParams, logger) => {
+module.exports.emptyBucket = async (objParams,) => {
```
Comment (simlu) [inline src/logic/util/s3.js]:
```suggestion
module.exports.emptyBucket = async (objParams) => {
```

File: src/logic/util/s3.js [hunk 12]
```diff
@@ -14,28 +15,26 @@ module.exports.putGzipObject = (bucket, key, data) => aws.call('s3:putObject', {
...
-          Bucket: objParams.Bucket,
-          ContinuationToken: result.NextContinuationToken
-        }, logger);
-      }
-      return Promise.resolve();
+      ContinuationToken: result.NextContinuationToken
     });
-  });
+  }
+  return Promise.resolve();
```
Comment (simlu) [inline src/logic/util/s3.js]: rm?

File: src/logic/util/s3.js [hunk 13]
```diff
@@ -14,28 +15,26 @@ module.exports.putGzipObject = (bucket, key, data) => aws.call('s3:putObject', {
...
+    Delete: {
+      Objects: objectList
     }
-    const objectList = result.Contents.map((c) => ({ Key: c.Key }));
-    logger.info(`Deleting ${objectList.length} items...`);
-    return deleteObjects({
+  });
+  logger.info(`Deleted ${data.Deleted.length} items ok.`);
+  if (result.IsTruncated) {
+    return this.emptyBucket({
```
Comment (simlu) [inline src/logic/util/s3.js]:
```suggestion
    await this.emptyBucket({
```

File: src/logic/util/s3.js [hunk 14]
```diff
@@ -14,28 +15,26 @@ module.exports.putGzipObject = (bucket, key, data) => aws.call('s3:putObject', {
...
 
-module.exports.emptyBucket = (objParams, logger) => {
+module.exports.emptyBucket = async (objParams) => {
   logger.info(`emptyBucket(): ${JSON.stringify(objParams)}`);
-  return listObjectsV2(objParams).then((result) => {
-    if (result.Contents.length === 0) {
-      return Promise.resolve();
+  const result = await listObjectsV2(objParams);
+  if (result.Contents.length === 0) {
+    return Promise.resolve();
```
Comment (simlu) [inline src/logic/util/s3.js]:
```suggestion
    return;
```

File: src/logic/process-logs.js [hunk 15]
```diff
@@ -69,23 +99,29 @@ const getToLog = async (resultParsed, rb) => {
...
-      rb[logLevel](processedLogEvent, process.env.ENVIRONMENT);
+      await Promise.all([
+        s3.putGzipObject(
+          process.env.LOG_STREAM_BUCKET_NAME,
+          `${resultParsed.logGroup.slice(1)}/${year}/${month}/${day}/${logLevel}-${logEvent.id}.json.gz`,
+          JSON.stringify(processedLogEvent)
+        ),
+        postToRollbar({
+          level: logLevel,
+          message: processedLogEvent.message,
```
Comment (simlu) [inline src/logic/process-logs.js]: This should also contain all of `processedLogEvent`?

**Pull request merged**